File: src/gestorAplicacion/clasesPrincipales/Ejercicio.java
package gestorAplicacion.clasesPrincipales;
import gestorAplicacion.clasesHerencia.*;
import gestorAplicacion.clasesEnum.*;
//import java.io.Serializable;

import java.util.Random;


public class Ejercicio extends Movimiento /*implements Serializable*/ {
	private static final long serialVersionUID = 1L;
	private int repeticiones;
	private int series;
	
	public Ejercicio(String nombre, String descripcion, String musculoPrincipal, TipoEjercicio tipoEjercicio, Maquina maquina, int repeticiones, int series, boolean ejercicio) {
		super(nombre, descripcion, musculoPrincipal, tipoEjercicio, maquina, null,  ejercicio);
		this.setRepeticiones(repeticiones);
		this.setSeries(series);
	}
	public Ejercicio(String nombre, String descripcion, String musculoPrincipal, TipoEjercicio tipoEjercicio, Maquina maquina, int repeticiones, int series, NivelCliente dificultad) {
		super(nombre, descripcion, musculoPrincipal, tipoEjercicio, maquina, dificultad);
		this.setRepeticiones(repeticiones);
		this.setSeries(series);
	}

	public int getRepeticiones() {
		return repeticiones;
	}

	public void setRepeticiones(int repeticiones) {
		this.repeticiones = repeticiones;
	}

	public int getSeries() {
		return series;
	}

	public void setSeries(int series) {
		this.series = series;
	}
	
	public static Ejercicio generarEjercicio(Movimiento movimiento, ObjetivoCliente objetivo) {
		int repeticiones = 0;
		int series = 0;
		Random random = new Random();

		if (movimiento.getTipoEjercicio() == TipoEjercicio.CALENTAMIENTO){
			repeticiones = 10 +random.nextInt(5);
			series = 1;
			movimiento.setDescripcion(movimiento.getDescripcion() + "	|Recuerda que los calentamientos están expresados en minutos");
		}else {
			switch(objetivo){
				case AUMENTAR:
					repeticiones = 8 + random.nextInt(4);
					series = 1 + random.nextInt(2);
					break;
				case DEFINIR:
					repeticiones = 14 + random.nextInt(3);
					series = 3 + random.nextInt(2);
					break;
				case BAJARPESO:
					repeticiones = 9 + random.nextInt(5);
					series = 2 + random.nextInt(2);
					break;
				case MANTENER:
					repeticiones = 11 + random.nextInt(2);
					series = 2 + random.nextInt(2);
					break;
				case ACONDICIONAR:
					repeticiones = 7 + random.nextInt(3);
					series = 4 + random.nextInt(2);
					break;
				
			}
		}
		return new Ejercicio(movimiento.getNombre(), movimiento.getDescripcion(), movimiento.getMusculoPrincipal(), movimiento.getTipoEjercicio(), movimiento.getMaquina(), repeticiones, series, false);
	}
}

File: Python/src/gestorAplicacion/clasesPrincipales/Ejercicio.py
from src.gestorAplicacion.clasesHerencia.Movimiento import Movimiento
from src.gestorAplicacion.clasesEnum import TipoEjercicio, NivelCliente, ObjetivoCliente
from random import randint

# Falta documentar Ejercicio


class Ejercicio(Movimiento):
    def __init__(self, nombre, tipo_ejercicio, maquina, repeticiones, series, dificultad: NivelCliente = None):
        super().__init__(nombre, tipo_ejercicio, maquina, dificultad)
        self.repeticiones = repeticiones
        self.series = series

    @classmethod
    def generarEjercicio(self, movimiento, objetivo):
        repeticiones = 0
        series = 0

        if movimiento.tipo_ejercicio == TipoEjercicio.CALENTAMIENTO:
            repeticiones = 10 + randint(0, 4)
            series = 1
            movimiento.descripcion += "	|Recuerda que los calentamientos están expresados en minutos"
        else:
            if objetivo == ObjetivoCliente.AUMENTAR:
                repeticiones = 8 + randint(0, 3)
                series = 2 + randint(0, 1)
            elif objetivo == ObjetivoCliente.DEFINIR:
                repeticiones = 14 + randint(0, 2)
                series = 3 + randint(0, 1)
            elif objetivo == ObjetivoCliente.BAJARPESO:
                repeticiones = 9 + randint(0, 4)
                series = 2 + randint(0, 1)
            elif objetivo == ObjetivoCliente.MANTENER:
                repeticiones = 11 + randint(0, 1)
                series = 2 + randint(0, 1)
            elif objetivo == ObjetivoCliente.ACONDICIONAR:
                repeticiones = 7 + randint(0, 2)
                series = 4 + randint(0, 1)

        return Ejercicio(movimiento.nombre, movimiento.tipo_ejercicio, movimiento.maquina, repeticiones, series,
                         movimiento.dificultad)


def generarEjercicio(mo1, ACONDICIONAR):
    return None